eng/tools/typespec-validation/src/rules/tspconfig-common-az-service-dir-match-pattern.ts

```diff
@@ -0,0 +1,12 @@
+import { TspconfigParameterRuleBase } from "../tspconfig-rule-helper.js";
+
+export class TspConfigCommonAzServiceDirMatchPatternRule extends TspconfigParameterRuleBase {
+  constructor() {
+    super(
+      "tspconfig-common-az-service-dir-match-pattern",
+      'Validate "parameters.service-dir.default".',
+      "service-dir",
+      /^sdk\/[^\/]*$/,
```

Comment:
Is there a reason you used `new RegExp()` above but not here?

Reply:
I'm not sure what you are saying, could you rephrase your comment?